### ISSUE TITLE
Editing a user

### ISSUE THREAD

**Issue by RyanTG** [scottwainstock/pbm#548]
Right now on live, super admins can edit others users (note that currently all users are admins). The edit screen looks like this:

![selection_012](https://cloud.githubusercontent.com/assets/632682/20913387/505f9bf8-bb2c-11e6-9b25-5ee06428588d.png)

I can change their email address, password, region, some privileges, and delete them. 

On auth, this edit screen only has an "is disabled" field:

![selection_015](https://cloud.githubusercontent.com/assets/632682/20913427/ae10c2f4-bb2c-11e6-8a60-b600367ba55f.png)

(also, this is potentially a bug: there is no "is disabled" field on this user: /admin/user/81/edit)

Going through these items one by one:

- Update email address

We give ANY user (admins and users) the ability to update their email address, so I guess we don't NEED to give the super admin the ability to change everyone's email address.

- password

Similarly, We give ANY user (admins and users) the ability to update their password, so I guess we don't NEED to give the super admin the ability to change everyone's password.

- region

This one seems necessary. As it is, I don't how I would create a regional admin. So I think we need this dropdown menu. When an admin steps down from being an administrator, we need to be able to return to a normal user (rather than disable them completely). 

Speaking of creating an admin, right now I just create a user, assign them to a region, and give them a password. In the future, I think we'll have them create a user themselves, and then we'll assign them to a region via this screen. Sound ok?

- Is machine admin

Doesn't seem necessary

- Is primary email contact

I don't think this is relevant anymore.

- delete

We're disabling users instead of deleting them. So we don't need delete.

- Edit username

This would be a new field. [See my comment here](https://github.com/scottwainstock/pbm/issues/549#issuecomment-265216545). Admins can edit their own username. Super admins can edit anyone's username.

**To summarize:** On this user edit screen, we should to add:

For SUPER admin:
- a region dropdown
- an is_disabled button when editing admin accounts (currently is_disabled is only visible when editing a regular user).
- "update username" field

For ALL admins:
- "update username" field only to their own username
- is_disabled button when editing a regular user

**Comment by RyanTG** [comment]
It looks like these options SHOULD be available already to the super user: https://github.com/scottwainstock/pbm/blob/auth_required/config/initializers/rails_admin.rb#L379

This begs the question, is gratzer@gmail.com not a super user on staging? It seems like I am, because I can see Banned IPs, Machines, Machine Groups.

**Comment by RyanTG** [comment]
I made some edits to the end of that comment.

I was trying to do this on my own. I was able to add an "update username" field visible to super user (I didn't try it out though). And I was nearly able to add a region dropdown. The name "region" showed up, but the dropdown wasn't visible. I think I'll leave that stuff to you.

**Comment by RyanTG** [comment]
After looking at ability.rb, I thiiiink my user might not be the super admin. If that's the case, that would change some of these comments.

Can you tell me if gratzer@gmail.com is a super admin on staging?

**Comment by scottwainstock** [comment]
I think you have access to the staging database, right?

heroku pg:psql -a pinballmapstaging

select * from users where email='gratzer@gmail.com';

**Comment by RyanTG** [comment]
I logged in with your zit email (guessed pw). I was NOT logged in as a super user. Going to review and then edit this issue.

**Comment by RyanTG** [comment]
Closing. I'm a kook.

**Comment by RyanTG** [comment]
The only thing not addressed is that regular admins can't edit their own username. We'll discuss about https://github.com/scottwainstock/pbm/issues/549 and see if it should be added for them. And if so, it'll be a fresh issue.
